### ISSUE TITLE
[entropy_src/rtl] optional new drop points when entropy overflows FIFOs

### ISSUE THREAD

**Issue by mwbranstad** [lowRISC/opentitan#6084]
Requirements have changed such that the drop point where new entropy cannot be consumed by entropy_src should be changed. A proposal would be to make this selectable (config bits). The choices would be the current input to the FIFO chain, the tail of the final FIFO (keep prior entropy seeds), and the head of the final FIFO (always push old entropy seeds out).

The attached diagram shows where these points are in the entropy flow.
[drop_point_diag.pdf](https://github.com/lowRISC/opentitan/files/6305109/drop_point_diag.pdf)

**Comment by mwbranstad** [comment]
For entropy meeting discussion: @tjaychen @martin-lueker @cdgori @felixmiller @moidx

**Comment by mwbranstad** [comment]
forget @senelson7 @zi-v

**Comment by mwbranstad** [comment]
Meeting consensus was that this should not be programmable, but rather just fixed, and that option #2 was preferred. Since this is trivial, that will be the direction of this issue.

**Comment by zi-v** [comment]
Does "The choices would be the current input to the FIFO chain" drop point mean after the health tests and before the '2kB' FIFO?

**Comment by tjaychen** [comment]
not quite. If you click on mark's diagram, you'll see that drop point 2 is after the final FIFO.  Which means data will get healthtested, conditioned via the sha3 and deposited into the FIFO.  An older entry in the FIFO will simply just be evicted.

@mwbranstad let me know if i misunderstood.

**Comment by mwbranstad** [comment]
The current drop point is at the very first FIFO, before the health tests. The proposed change is drop point #2, as @tjaychen has properly described.

**Comment by mwbranstad** [comment]
[drop_point_diag2.pdf](https://github.com/lowRISC/opentitan/files/6378708/drop_point_diag2.pdf)

Updated drawing based on discussions.

**Comment by tjaychen** [comment]
@zi-v 
thanks Mark, this looks good to me.

**Comment by zi-v** [comment]
LGTM as well. Thanks.

**Comment by cdgori** [comment]
As we discussed this morning, LGTM.
